ui/apps/platform/src/Containers/NetworkGraph/NetworkGraph.tsx

```diff
@@ -77,19 +77,49 @@ function getUrlParamsForEntity(selectEntity: {
 }
 
 export type NetworkGraphProps = {
-    detailType?: UrlDetailTypeValue;
-    detailId?: string;
     model: Model;
 };
 
 export type TopologyComponentProps = NetworkGraphProps;
 
-const TopologyComponent = ({ detailType, detailId, model }: TopologyComponentProps) => {
+function getNodeEdges(selectedNode) {
+    const egressEdges = selectedNode.getSourceEdges();
+    const ingressEdges = selectedNode.getTargetEdges();
+    return [...egressEdges, ...ingressEdges];
+}
+
+function setVisibleEdges(edges) {
+    edges.forEach((edge) => {
+        edge.setVisible(true);
+    });
+}
+
+const TopologyComponent = ({ model }: TopologyComponentProps) => {
+    const { detailType, detailId } = useParams();
```

Comment:
Wouldn't the fix still work if `useParams` was in the parent component? I think it makes sense to keep `useParams` in the parent component since we don't want this component to start doing too many things. Maybe you can try testing that theory out and seeing if it anything happens

Reply:
okay turns out we do need to keep it at this level to prevent rerenders of the parent component on each node select/unselect. it was still flickering when i moved it out and post rebase from moving `history` out as well. when i moved them both back in as it was in this PR initially, the flickering went away